### PR TITLE
Fix date time range check with correct TZ offset

### PR DESCRIPTION
**Pull request by barwi**

### CODE REVIEW

**Comment by yurikuzn** [comment]
Any info?

**Comment by barwi** [comment]
DateTime operators must use correct conversion between localtime (from request passed with date part only) and UTC. Therefore TZ offset shoud be applied to both sides of the date range: day start and day end.

For example, converting user's last Sunday of March or October (when daylight savings time begin/end) to UTC timestamp must produce range 23 or 25 hours length.

File: application/Espo/Core/SelectManagers/Base.php [hunk 1]
```diff
@@ -959,6 +962,7 @@ public function convertDateTimeWhere($item)
             case 'after':
                 $where['type'] = 'after';
                 $dt = new \DateTime($value, new \DateTimeZone($timeZone));
+                $dt->modify('+1 day -1 second');
```
[the remark this answers is not in the document]
Comment (barwi) [reply application/Espo/Core/SelectManagers/Base.php]: Condition "after selected day" must not include any second of this day (after 23:59:59 in local time). It's shoud work like "before" (before 00:00:00 local).

[the remark this answers is not in the document]
Comment (yurikuzn) [reply application/Espo/Core/SelectManagers/Base.php]: Sure, you're right.

[the remark this answers is not in the document]
Comment (yurikuzn) [reply application/Espo/Core/SelectManagers/Base.php]: Thanks for help.

[the remark this answers is not in the document]
Comment (yurikuzn) [reply application/Espo/Core/SelectManagers/Base.php]: I believe $dt->modify('+1 day -1 second'); should be before applying timezone.

**Comment by yurikuzn** [comment]
Unfortunately, this fix brought about serious problems. It doesn't work properly date-time values. 3rd party calendar integration stopped to work. I will fix it.

**Pull request closed without merging**